### ISSUE TITLE
bug(useminprepare): crashes when looking for a build script HTML comment blocks

### ISSUE THREAD

**Issue by pgilad** [yeoman/grunt-usemin#246]
crash reports error:
`Warning: An error occurred while processing a template (Unexpected identifier). Use --force to continue.`

This happened only when I updated my useMin task to the latest version. Used to run fine before.

Running task with --verbose or --debug doesn't add any information.
Running task with --stack reports error is in lodash function.

Tracing the error through code myself I found the following erroneous block:

```
  var config;
            try {
                config = c.process(filepath, grunt.config());
            }
            catch (e) {
                grunt.fail.fatal(e);
            }
```

The actual problem (which wasn't caught in the try catch) is `grunt.config()` is being passed instead of just `grunt.config`.
Changing the passed param to just `grunt.config` solved the issue for me.

npm version:
node: 0.10.15
usemin: 2.0.2

my useminPrepare config:

```
  useminPrepare: {
            html   : ['<%= config.src %>/options.html', '<%= config.src %>/background.html'],
            options: {
                dest: '<%= config.dist %>'
            }
        },
```

**Comment by bbBobbyEarl** [comment]
I was also experiencing this issue.  In my initConfig I'm reading a YAML file.  In that YAML file I use lo-dash templates to reference other properties in the YAML as described by http://assemble.io.  I had a YAML variable referencing itself, like:

`myVar: <%= myVar %>/asdf`

I believe this was causing the grunt.config() function to never return.  Luckily it seems the ConfigWriter library can handle being passed grunt.config or grunt.config().  My suggestion if you are also experiencing this error is try to track down why grunt.config() wouldn't be returning.

**Comment by korto** [comment]
@bbBobbyEarl indeed.

**Comment by stephanebachelier** [comment]
@korto this issue has been closed by the author. I let it closed as it seems to be a duplicate of #291.
